src/topics/inputs.md

```diff
@@ -268,28 +269,35 @@ that accepts `null` (i.e. no value provided), or any value from an enum.
 :name: exclusive-parameter-expressions.cwl
 ```
 
-Note how the JavaScript expression uses the value of the exclusive input parameter
-without taking into consideration a `null` value. If you provide a valid value,
-such as “fasta” (one of the values of the enum), your command should execute
-successfully:
+Note how the JavaScript expression uses the value of the exclusive input
+parameter without taking into consideration a `null` value. If you provide a
+valid value, such as `fasta` (one of the possible values of the enum), your
+command should execute successfully:
 
 ```{runcmd} cwltool exclusive-parameter-expressions.cwl --file_format fasta
 :working-directory: src/_includes/cwl/inputs
-````
+```
 
 However, if you do not provide any input value, then `file_format` will be
-evaluated to a `null` value, which does not match the expected type for the
+evaluated to `null`, which does not match the expected type for the
 output field (a `string`), resulting in failure when running your workflow.
 
 ```{runcmd} cwltool exclusive-parameter-expressions.cwl
 :working-directory: src/_includes/cwl/inputs
 :emphasize-lines: 5-10
 ```
 
-To correct it, you must remember to use an or operator in your JavaScript expression
-when using exclusive parameters, or any parameter that allows `null`. For example,
-the expression could be changed to `$(inputs.file_format || 'auto')`, to have
-a default value if none was provided in the command line or job input file.
+To correct it, you should explicitly handle the possibility of a `null` value.
+For example, the expression could be changed to `$(inputs.file_format ||
+'auto')`, to have a default value `"auto"` if none was provided in the command
+line or job input file.
+
+Here, the boolean “or” operator `||` in JavaScript is used for its
+_short-circuiting_ property. If `inputs.file_format` takes a “truthy” value
+(e.g. a valid string from the enum), the evaluation of the expression stops at
+the first operand of `||`; it “short-circuits”. If however `inputs.file_format`
+is `null` -- a “falsy” value, the whole expression’s value becomes that of
```

Comment:
-- a “falsy” value -- ?

Reply:
I only meant the `--`. It looked to me like the `--` was a pause to indicate that `null` was considered a falsy value, and then we continued with "the whole expression's...".

I think your current explanation looks good! JS is indeed not consistent, but I guess that's one thing that most JS devs learn (probably the harder way through bugs :sweat_smile: )